parsel/selector.py

```diff
@@ -489,8 +489,8 @@ def remove(self) -> None:
             )
 
         try:
-            parent.remove(self.root)
-        except AttributeError:
+            self.root.drop_tree()
```

Comment:
I wonder if instead of changing selector.remove() to use element.drop_tree() instead of element.remove(), we should add a selector.drop_tree() shortcut, which will be using element.drop_tree().

Reply:
@kmike The issue with the current implementation is described here: 
- https://github.com/scrapy/parsel/issues/215

Maybe for this situation is better to modify instead of create new method?

Reply:
@kmike Thoughts?

Reply:
I'm not sure; it'd be interesting to hear what others think.

The trade-off is this: 

1) either we follow lxml method names (.remove and .drop_tree), or 
2) we try to have better names - have Selector.remove behave like .drop_tree, not like lxml's .remove.

Following lxml names could be less surprising for people familiar with lxml, and it'd also be backwards compatible. Changing .remove behavior could be less surprising for people in general, but could create some confusion for people who use lxml, and it's a backwards incompatible change.

Both approaches look fine to me.

Reply:
I can refactor to approach 1, i think will be better for backward compatibility.

Reply:
I see [the fact that the trailing text is part of an element comes from the standard library](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.tail), and [the way they document `drop_tree()`](https://lxml.de/lxmlhtml.html#html-element-methods), I think lxml aimed to work around precisely this issue.

I have no strong opinion here.

I see no use for `delete()` as lxml’s `.getparent().delete()`, though. So I would not keep both `delete` and `drop_tree`. I think we either change the behavior of `delete()`, and I think that may be the best approach here, or we leave `delete` as is, we deprecate it, and we implement the new, desired behavior as `drop_tree` (or as `drop`, `pop`, whatever, I do not think we need to map lxml naming here specially since this feature we may want to eventually support for non-lxml content).

Reply:
What you think about add a new parameter in `.remove`?

Like:

```
remove(drop_tree: bool = False)
```

This way we dont break current behavior.

Reply:
`delete` does remove a whole tree, though. We could do `remove(keep_trail: bool = False)`. But still, people needing to pass that parameter for the most common use case feels wrong.

Reply:
I will let this decision to you guys. Whatever approach choosed i can apply. 🤓

Reply:
For some reason when i request review one of reviewer is removed, github bugs...

Reply:
>  see no use for delete() as lxml’s .getparent().delete()

I guess the use case for the current behavior is the following: you get all children from a node, and calling .delete for them on-by-one. In the end, the content of the node is empty. With drop_tree the text which is in tails will remain. It looks a bit obscure though.

Reply:
hey @Laerte! We discussed it internally, and finally agreed on the approach :)

1. Keep the behavior of Selector.remove, but deprecate this method.
2. Keep the behavior of SelectorList.remove, but deprecate it.
3. Introduce a new Selector.drop method, with the new behavior.
4. Introduce a new SelectorList.drop method, with the new behavior.

The reason to keep .remove as-is is to avoid breaking changes to the user code.
The reason a new method is called .drop, and not .drop_tree is that we want the method name to be the same in Selector and SelectorList, and for drop_tree it'd probably make more sense to call it SelectorList.drop_tree**s**, which is an issue.

Reply:
@kmike Thanks for the feedback, i will work on this later.

Reply:
We also decided that it's not a blocker for the upcoming release so no hurry regarding that.

Reply:
I did the first draft, what you guys think? About documentation we will replicate the `remove` part but using drop?

Reply:
Yes, you can basically update https://parsel.readthedocs.io/en/latest/usage.html#removing-elements to use `drop` instead (for the code parts, maybe it makes sense to keep using the English word “remove” for non-code parts), as if `remove` had never existed.